KeyDerivationInterface/Keys/Keys.cs

```diff
@@ -1,213 +1,214 @@
+    {
+        private const string BitcoinEllipticCurveName = "secp256k1";
+        private DerObjectIdentifier curveOid;
+        private ECKeyGenerationParameters keyParams;
+
+        private readonly ECMultiplier multiplier = new FixedPointCombMultiplier();
+
+        private const int KeyChainCodeLength = 32;
+        private ECPoint publicKey;
+        private byte[] chainCode;
+
+        private PublicDerivationKey()
+        {
+            curveOid = ECNamedCurveTable.GetOid(BitcoinEllipticCurveName);
+            keyParams = new ECKeyGenerationParameters(curveOid, new SecureRandom());
+        }
+
+        public PublicDerivationKey(byte[] privateKey, byte[] chainCode) : this()
+        {
+            PublicKey = multiplier.Multiply(keyParams.DomainParameters.G, new Org.BouncyCastle.Math.BigInteger(1, privateKey)).Normalize();
+            ChainCode = chainCode;
+        }
+
+        public PublicDerivationKey(ECPoint publicKey, byte[] chainCode) : this()
+        {
+            PublicKey = publicKey;
+            ChainCode = chainCode;
+        }
+
+        public ECKeyGenerationParameters KeyParams
+        {
+            get => keyParams;
+        }
+
+        public ECPoint PublicKey
+        {
+            get
+            {
+                return publicKey;
+            }
+
+            internal set
+            {
+                publicKey = value ?? throw new KeyCreationException($"Derivation public key can not be a null.");
+            }
+        }
+
+#pragma warning disable CA1819 // Properties should not return arrays
+
+        public byte[] ChainCode
+        {
+            get
+            {
+                return chainCode;
+            }
+
+            internal set
+            {
+                if (value == null)
+                {
+                    throw new KeyCreationException($"Derivation key chain code can not be a null.");
+                }
+
+                if (value.Length != KeyChainCodeLength)
+                {
+                    throw new KeyCreationException($"Derivation key chain code length should be equal to {KeyChainCodeLength} bytes.");
+                }
+
+                chainCode = value;
+            }
+        }
+#pragma warning restore CA1819 // Properties should not return arrays
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as PublicDerivationKey);
+        }
+
+        public bool Equals(PublicDerivationKey other)
+        {
+            if (this == other)
+            {
+                return true;
+            }
+            if (other == null)
+            {
+                return false;
+            }
+
+            return PublicKey.Equals(other.PublicKey) &&
+                   ChainCode.SequenceEqual(other.ChainCode);
+        }
+
+        public override int GetHashCode()
+        {
+            return publicKey.GetHashCode() + new BigInteger(ChainCode).GetHashCode();

```

Comment:
Please use HashCode.Combine or (publicKey, BigInteger(ChainCode)).GetHashCode()

Reply:
Struct HashCode appears in .NET Standard 2.1 only.
If  I use (publicKey, BigInteger(ChainCode)).GetHashCode() it returns different values of HashCode all the time I start tests for the same object. I think it's a wrong behavior for GetHashCode() function.